### PR TITLE
[feature] #49: Add the graceful shutdown for daemon mode

### LINKED ISSUE #49: Graceful shutdown on `daemon` mode
[Currently](https://github.com/soramitsu/iroha2-longevity-load-rs/pull/38/files#diff-2e34ce8e60a784d945b36d0cefcbd4e8f926330c8636a60650e0f3dfeea46415R100), `daemon` mode uses multiple spawned tasks, and all these should stop without forcing.

### PR DESCRIPTION
**Pull request by pesterev**
Signed-off-by: Vladimir Pesterev <pesterev@pm.me>

# Task

Closes #49 

## Changes

1. Add the graceful shutdown to the `daemon` mode.

## Author

Signed-off-by: Vladimir Pesterev <pesterev@soramitsu.co.jp>

### CODE REVIEW

File: src/commands/daemon.rs [hunk 1]
```diff
@@ -153,14 +175,18 @@ fn perform_operations(
...
     mut operations: HashMap<Operation, usize>,
+    is_closed: Arc<AtomicBool>,
 ) {
     let alice_id = AccountId::from_str("alice@wonderland").expect("Failed to make Alice id");
     let wonderland_id =
         DomainId::new(Name::from_str("wonderland").expect("Failed to create Wodnerland name"));
     while !operations.is_empty() {
+        if is_closed.load(Ordering::SeqCst) {
+            return;
+        }
```
Comment (0x009922) [inline src/commands/daemon.rs:182]: Am I right that the daemon process will ignore <kbd>CTRL + C</kbd> until it receives an event from Iroha?

Is there any way to stop listening for events immediately?

Comment (pesterev) [reply src/commands/daemon.rs:182]: It seems, unfortunately, the client library doesn't provide a way to listen for events without blocking or with the option to cancel blocking listening.

My suggestion is to add something like a cancellation token to `listen_for_events` in the client library.

I've opened an issue for that.
https://github.com/hyperledger/iroha/issues/2940

Comment (pesterev) [reply src/commands/daemon.rs:182]: Anyway, we can keep open this pull request while [that suggestion](https://github.com/hyperledger/iroha/issues/2940) is being implemented.

Comment (appetrosyan) [reply src/commands/daemon.rs:182]: This is a really good idea. I'd suggest drafting this PR and adding it as blocked on zenhub @pesterev

**Pull request closed without merging**